### PR TITLE
Should not re-render as the result of a null state update

### PR DESCRIPTION
**Pull request by acdlite**
Fiber already happens to behave this way; this just adds a test.

~~TODO: Should we add this feature to Stack, too?~~ Done.

See this post for context: https://www.facebook.com/groups/2003630259862046/permalink/2190298087861928/

### CODE REVIEW

File: src/renderers/shared/shared/__tests__/ReactUpdates-test.js [hunk 1]
```diff
@@ -13,9 +13,9 @@
 
 var React;
 var ReactDOM;
-var ReactDOMFeatureFlags;
 var ReactTestUtils;
 var ReactUpdates;
+var ReactDOMFeatureFlags = require('ReactDOMFeatureFlags');
```
Comment (sebmarkbage) [inline src/renderers/shared/shared/__tests__/ReactUpdates-test.js]: Why is this necessary? We don't do this elsewhere do we?

Comment (acdlite) [inline src/renderers/shared/shared/__tests__/ReactUpdates-test.js]: Uh... I don't remember. I thought I did this in a file where the branch wrapped a `describe` block but that's not true of this file.
